src/main/java/pl/xkoem/OXGame.java

```diff
@@ -0,0 +1,45 @@
+package pl.xkoem;
+
+import pl.xkoem.gamestates.EndOfGame;
+import pl.xkoem.gamestates.Init;
+import pl.xkoem.gamestates.Match;
+
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public class OXGame {
+
+    private Players players;
+    private final Supplier<String> userInput;
+    private final Consumer<String> userOutput;
+    private GameConfiguration gameConfiguration;
+
+    public OXGame(Players players, Supplier<String> userInput, Consumer<String> userOutput) {
+        this.players = players;
+        this.userInput = userInput;
+        this.userOutput = userOutput;
+        gameConfiguration = new GameConfiguration();
+    }
+
+
+    /**
+     * Główna metoda uruchamiająca cały proces gry po kolei
+     */
+
+    public void run() {
+        Init init = new Init(userInput, userOutput, players, gameConfiguration);
```

Comment:
Linijki od tej do 34 wyglądają jak jeden wielki init(), to możesz przerzucić to do takiej metody